python/pyspark/sql/context.py

```diff
@@ -427,6 +428,31 @@ def func(iterator):
             df = self._ssql_ctx.jsonRDD(jrdd.rdd(), scala_datatype)
         return DataFrame(df, self)
 
+    def range(self, start, end, step=None, num_partitions=None):
+        """Creates a [[DataFrame]] with a single [[LongType]] column named `id`, containing
+        elements in an range from `start` to `end`(exclusive) with an Integer step value, with
+        partition number specified.
+        """
+        step = 1 if step is None else step
+        num_partitions = self._sc.defaultParallelism if num_partitions is None else num_partitions
+        if not isinstance(start, (int, long)):
+            raise TypeError("start should be long")
```

Comment:
There is no `long` in Python 3, so we should do this (accept long in python 2):
```
if not isinstance(start, (int, long)):
  raise TypeError("start should be int")
```

Python 2 will convert value larger than sys.maxint to long.